connectors/webhook/src/test/java/io/camunda/connector/inbound/authorization/JWTIntegrationTest.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.inbound.authorization;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.catchThrowable;
+import static org.junit.Assert.assertEquals;
+
+import com.auth0.jwk.Jwk;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwk.JwkProviderBuilder;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTCreator;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.common.ClasspathFileSource;
+import io.camunda.connector.api.inbound.webhook.WebhookProcessingResult;
+import io.camunda.connector.impl.inbound.result.MessageCorrelationResult;
+import io.camunda.connector.inbound.HttpWebhookExecutable;
+import io.camunda.connector.inbound.model.WebhookConnectorProperties;
+import io.camunda.connector.inbound.utils.ObjectMapperSupplier;
+import io.camunda.connector.inbound.utils.TestRSAKeyProvider;
+import io.camunda.connector.test.inbound.InboundConnectorContextBuilder;
+import io.camunda.connector.test.inbound.InboundConnectorPropertiesBuilder;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+@Disabled // to be run manually
```

Comment:
Just curious: does it take too much time to run in a CI environment?

Reply:
Not at all, only a few seconds. Reason I disabled is because I would like to avoid to lunch a mock API server when we run the tests in GitHub actions. We have to specify to port on which we would like this server to listen and this port might be occupied already which would lead to false test fails.

Reply:
@markfarkas-camunda I think its great to have this test. Lets try to enable it. You can use https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/context/SpringBootTest.WebEnvironment.html#RANDOM_PORT to use a random port for the server and you can use 
```
@LocalServerPort
private Integer port;
``` to inject it.

For wiremock you can use `.dynamicPort()` to use a random port for the mocking server: https://wiremock.org/docs/configuration/